coala_quickstart/generation/Project.py

```diff
@@ -50,6 +52,22 @@ def language_percentage(file_paths):
         if ext in exts:
             for lang in exts[ext]:
                 results[lang] += delta
+        elif not ext:
+            with open(file_path, 'r') as data:
+                hashbang = data.readline()
+                if(re.match(
+                        '/(^#![(a-z)|\/]*[ ][a-z]*)|(^#![(a-z)|\/]*)/',
+                        hashbang)):
+                    language = get_language_from_hashbang(hashbang)
+                    try:
+                        for ext in exts:
+                            for lang in exts[ext]:
+                                if language == lang.lower():
+                                    results[lang] += delta
+                    except KeyError:
+                        printer.print("Cannot recognize the language",
+                                      color="red")
+                    data.close()
```

Comment:
close only if the `if` returns true? ;) what if there isn't a regex match?

And you don't really need this anyway - the context manager will automatically close it for you when it goes out of scope.